### DISCUSSION TITLE
I need help with typescript

### DISCUSSION THREAD

**Discussion by BkNcs** [community/community#70727] [Actions]
### Select Topic Area

General

### Body

I cant' install typescript in my host. It's working in replit, but when I try to install the bot in my host:

![276864953-01aef1a4-b536-4c40-80b2-cbb5b64e7d80](https://github.com/community/community/assets/116917089/5b69fac2-76ee-4799-b4a7-0ad18b9bda8b)

Someone can help me?

**Comment by itsdotnet** [comment]
Please give more information like what is this app name
What do you want : coding ?
After that I think we can help you

**Reply by BkNcs** [reply]
I'm running a Discord bot made in typescript on a host with pterodactyl panel, it gives me an error that it can't run the file, I have the files like this:

package.json

>{
  "name": "discord-application-bot",
  "version": "1.0.0",
  "description": "A Discord Bot for accepting or rejecting users based on their form submissions",
  "main": "./src/index.ts",
  "type": "module",
  "scripts": {
    "test": "echo \"Error: no test specified\" && exit 1",
    "debug": "ts-node --esm --trace-warnings ./src/index.ts",
    "start": "ts-node --esm ./src/index.ts"
  },
  "repository": {
    "type": "git",
    "url": "git+https://github.com/JuanDelPueblo/discord-application-bot.git"
  },
  "keywords": [
    "discord",
    "whitelist",
    "application",
    "forms",
    "bot",
    "forms",
    "moderation"
  ],
  "author": "JuanDelPueblo",
  "license": "MIT",
  "bugs": {
    "url": "https://github.com/JuanDelPueblo/discord-application-bot/issues"
  },
  "homepage": "https://github.com/JuanDelPueblo/discord-application-bot#readme",
  "dependencies": {
    "@json2csv/plainjs": "^7.0.1",
    "discord-api-types": "^0.37.49",
    "discord.js": "^14.11.0",
    "install": "^0.13.0",
    "node": "^21.0.0",
    "node.js": "^0.0.1-security",
    "reflect-metadata": "^0.1.13",
    "sequelize": "^6.32.1",
    "sequelize-typescript": "^2.1.5",
    "sqlite3": "^5.1.6",
    "tsc": "^2.0.4"
  },
  "devDependencies": {
    "@types/eslint": "^8.44.0",
    "@types/node": "^20.4.2",
    "@types/validator": "^13.7.17",
    "eslint": "^8.45.0",
    "ts-node": "^10.9.1",
    "typescript": "^5.1.6"
  }
}

tsconfig.json

>{
	"compilerOptions": { /* Docs: [https://www.typescriptlang.org/tsconfig](https://www.typescriptlang.org/tsconfig) */
        "esModuleInterop": true,
		"target": "ES2022",
		"module": "es2022",
		"outDir": "bin",
		"rootDir": "src",
		"strict": true,
		"noImplicitAny": true,
		"strictNullChecks": true,
		"moduleResolution": "NodeNext",
		},
    "ts-node": {
        "esm": true
        },
		"lib": [
			"es6"
		],
    }
		"esModuleInterop": true,
		"skipLibCheck": true,
		"forceConsistentCasingInFileNames": true,
		"allowUnreachableCode": true,
		"useUnknownInCatchVariables": false, // TypeScript 4.4+ only!
		"experimentalDecorators": true,
		"emitDecoratorMetadata": true,
	},
	"include": [
		"src/**/*.ts"
	],
	"exclude": [
		"node_modules",
		"<node_internals>/**",
               ]
}

**Comment by unicornware** [comment]
node was given a typescript file to run, rather than a file format it recognizes, i.e. javascript. 

if you want to use `node` to run typescript files, you'll need a [custom loader](https://nodejs.org/api/esm.html#loaders). an example can be seen [here](https://github.com/flex-development/grease/blob/main/loader.mjs). 

you can also try using `ts-node`, or building your project and running the resulting javascript files instead.
